commit cfa146e4be274fd04bfdb26b3c96cdfe81a43dc2
Author: Jouni Malinen <jkmaline@cc.hut.fi>
Date:   Fri Mar 24 21:24:55 2006 -0800

    [PATCH] hostap: Fix EAPOL frame encryption
    
    Fixed encrypted of EAPOL frames from wlan#ap interface (hostapd). This
    was broken when moving to use new frame control field defines in
    net/ieee80211.h. hostapd uses Protected flag, not protocol version
    (which was cleared in this function anyway). This fixes WPA group key
    handshake and re-authentication.
    http://hostap.epitest.fi/bugz/show_bug.cgi?id=126
    
    Signed-off-by: Jouni Malinen <jkmaline@cc.hut.fi>
    Signed-off-by: John W. Linville <linville@tuxdriver.com>

diff --git a/drivers/net/wireless/hostap/hostap_80211_tx.c b/drivers/net/wireless/hostap/hostap_80211_tx.c
index a881212e1fae..06a5214145e3 100644
--- a/drivers/net/wireless/hostap/hostap_80211_tx.c
+++ b/drivers/net/wireless/hostap/hostap_80211_tx.c
@@ -469,7 +469,7 @@ int hostap_master_start_xmit(struct sk_buff *skb, struct net_device *dev)
 	}
 
 	if (local->ieee_802_1x && meta->ethertype == ETH_P_PAE && tx.crypt &&
-	    !(fc & IEEE80211_FCTL_VERS)) {
+	    !(fc & IEEE80211_FCTL_PROTECTED)) {
 		no_encrypt = 1;
 		PDEBUG(DEBUG_EXTRA2, "%s: TX: IEEE 802.1X - passing "
 		       "unencrypted EAPOL frame\n", dev->name);